controllers/testclustertopology_controller.go

```diff
@@ -698,6 +703,104 @@ func (r *TestClusterTopologyReconciler) initTiUPResources(ctx context.Context, c
 	return requeue, nil
 }
 
+func (r *TestClusterTopologyReconciler) initTiDBResources(ctx context.Context, ct *naglfarv1.TestClusterTopology, resources []*naglfarv1.TestResource, clusterIPMaps map[string]string) (bool, error) {
+	var requeue bool
+	var haProxyResources []*naglfarv1.TestResource
```

Comment:
The spec only allows users fill one HAProxy host, why you create a slice to store the haproxy resource?

Reply:
Because haproxy supports high availability and is convenient for future expansion, we think slice can contain a single case